### DISCUSSION TITLE
🆕 Grafana Navigation Feedback

### DISCUSSION THREAD

**Discussion by zsofiakm** [grafana/grafana#58910] [General]
### What's new in Grafana v9.5
_**Grafana's new navigation is enabled by default from version 9.5.**_

Use Grafana’s redesigned navigation to get full visibility into the health of your systems, by quickly jumping between features as part of your incident response workflow.

As Grafana has grown from a visualization tool to an observability solution, we’ve added many tools along the way. This often resulted in pages that were visually inconsistent or hard to find. This update gives Grafana a new look and feel, and makes page layouts and navigation patterns more consistent.

We’ve revamped the navigation menu and grouped related tools together, making it easier to find what you need. Pages in Grafana now leverage new layouts that include breadcrumbs and a sidebar, allowing you to quickly jump between pages. We’ve also introduced a header that appears on all pages in Grafana, making dashboard search accessible from any page.

![image](https://user-images.githubusercontent.com/97596715/202470483-a656a99a-da3c-496b-a450-4ce6c46aa92a.png)

### Roadmap
Based on the feedback we've received, here are the improvements planned for subsequent releases:
- Making the contents of the main (hamburger) menu more easily accessible, to reduce the number of steps required to move between pages.
- UI improvements for the side navigation to support features with many subpages
- Layout improvements

### Feedback
Please let us know what you think:

- What did you find useful?
- Where challenges do you have with the new features?
- What are you still missing?
- Are there any bugs that you've found?
- Your feedback could help us decide what to add or change in a future iteration. ✨

**Comment by mari-arondeus** [comment]
Hi there! We're trying out the new Grafana topnav navigation feature, and I'm having trouble locating the LDAP browser previously located in the Grafana Admin tab. Is this missing intentionally?

**Reply by amy-super** [reply]
Hi @cassidy3 thanks for flagging this! Not at all intentional and we just noticed it internally as well 😅 
We'll be tracking the fix for in #60052 if you'd like to follow along. Until then, you can use the direct URL as a somewhat clunky workaround.

**Reply by mari-arondeus** [reply]
Thank you! Also one tiny other thing - could you provide the location of the SVG Grafana logo displayed in the new topnav? We were using custom docker mounts to rebrand the logo, but it doesn't appear as though the new nav is using the same file.

**Reply by amy-super** [reply]
@ashharrison90 is this something you can please help with?

**Reply by ashharrison90** [reply]
@cassidy3 thanks for pointing this out - that's a bug! have fixed it here: https://github.com/grafana/grafana/pull/60227 so you shouldn't have to change anything. 👍 it should be released in 9.4.0

**Reply by mari-arondeus** [reply]
Y'all are the best - thanks again!

**Comment by shaunmansell** [comment]
Hi, I've been testing out the topnav in Grafana 9.4 and thought I'd provide some feedback here on one navigation flow I hit challenges with.

I regularly use the import button under this sub-menu in the current navigation
<img width="220" alt="image" src="https://user-images.githubusercontent.com/21325844/225773785-cb9a23f3-4a69-455a-a9b6-59b238ce2bc0.png">
On trying to find this in the new nav, I struggled. I eventually found it under the "New" button on the `/dashboards` screen, though it was with a bit of dumb luck and going through a number of menus to eventually get there.

Could an import button be added to the new + button in the top right to bring back the same flow as the current nav?
e.g. under this
<img width="285" alt="image" src="https://user-images.githubusercontent.com/21325844/225774084-aaaad0c0-d719-48b8-a139-36877d7cb155.png">

That way users get immediate access to import from anywhere in Grafana and they don't have to navigate to a specific screen to then initiate import through 2 additional interactions.

**Reply by zsofiakm** [reply]
Hi @shaunmansell,
Thanks very much for the feedback, dashboard import will be available under the + button in the header from Grafana v9.5.

**Comment by ekostjuk** [comment]
Hi! Tested topnav feature, but had to raise a ticket to roll back.

Initially liked the consistency aspect(or rather - thought I did), but few things made me struggle. Got top-nav enabled in my personal account vs disabled in business one, which made the differences more obvious. Apologies, but mostly imho critics on this one:
- Extra clicks to navigate vs fast hover-pop-up flow. Looking at datasource - +2clicks, Looking at dashboards - +1 click. Looking at SynthMonitoring - +1 click. Added to all that +1 click to bring up the side panel for out of breadcrumb nav.
- Slow view re-renders. All aspects of UI are slow vs old snappy feel. Coupled with extra clicks - slows things down massively.
- Navigation causes context switches. Old: snappy pop-ups compact all aspects of nav into one screen area and you land where you need. If you are cross-checking dashboards for whatever reason - you'd likely still recollect one you navigated from easily. New: your eyes have to jump around screen to navigate and do extra clicks+load times. Screen scan+slow UI == context switch/stutter.
- I thought breadcrumbs are useful, but it feels more like patching a gap after removing a working tab interface. At max 2 depth levels and <10 tab items+pop-up from side panel - there is not much that can be done to make it better.
- Lost vertical space + extra top panel clutter. (Old dashboard section titles look really clean compared to it)
- Thought I liked icon consistency vs old weird mix of gray and colored ones(some gray, some orange, some unique), but eyes don't cling to these easily, forces an extra scan. I'd much prefer consistent style, but still uniquely color coded.
- Lack of pop-ups makes discovery of component sub-screen titles - a click-through-expands journey.
- Like Alert & RMI grouping side-view, but somehow pushes it down the UI tree, that you have to navigate out to move from alert to relevant dashboards. Maybe would need to use search bar more.

Overall - it feels that 'topnav' change undermines UX without solving issues of the old one.

Note: Provision everything with terraform, so ease of access to ClickOps features is on static dashboards, outside of playground is lost on me. (I'd never add alert manually) Also - feature showcase/marketing on component landing pages gets old quickly. Like - if I'll never use k6, why should I see the noise?

**Reply by zsofiakm** [reply]
Hi @ekostjuk,

We appreciate you taking the time to provide such detailed feedback, it's really helpful to hear about the challenges you're experiencing. We acknowledge that moving between some of the pages now takes more steps, and we're looking at making the contents of main menu more easily accessible accessible to address that.

Regarding the slow view re-renders you mentioned, we would appreciate if you could provide more specifics or examples. This will help us better understand the issue. 

Thanks again for your valuable input!

**Reply by amy-super** [reply]
Hi @ekostjuk - I'm interested in your feedback around "extra clicks." Would you be interested in joining me for a 45-minute interview to provide more context? Details are [here](https://www.userinterviews.com/projects/28-0kHr3pw/apply) in this sign-up link. (Edited because I forgot to mention that we'll share a $75 gift card with you as a thanks for your time if you're selected to participate.)

**Comment by meyerder** [comment]
How do I disable this.. as it is broken when it comes to Orgs and changing the orgs.. All it seems to do is change the org number..

**Comment by ashharrison90** [comment]
hi @meyerder 👋 

this will be the default navigation going forward so there will be no way to disable it. you can of course stay on an older version of grafana.

are you able to share more information about the behaviour you're seeing and your specific setup? it sounds like a bug. would be helpful to know how you're running grafana, what version of grafana, what os (and os version) you're on and what browser (and browser version) you're using.

thanks! 🙌

**Comment by meyerder** [comment]
Yes I have opened TT# 86521 which literally stops me dead in the tracks for leveraging 9.5 which I was hopeful would fix several other bugs in 9.4.x  without the ability to disable that I am stuck for yet another release

[page cant be found.zip](https://github.com/grafana/grafana/files/11240542/page.cant.be.found.zip)

**Comment by ashharrison90** [comment]
@meyerder i've altered the behaviour here to always redirect to the root page when switching organisation: https://github.com/grafana/grafana/pull/66655.

hope that helps!

**Comment by meyerder** [comment]
@ashharrison90 any reason why it can not do the same with teams that the member is on..  While I suspect a better means of doing this can be done I do not yet know how inside the DB how teams are stored yet.. So I built a view and made a panel called Teams. and display this.. I also force the forcelogin to happen in the event of a timeout or similar which directs them to the teams home page.. This comes in very handy why users are part of more than one org or team

SELECT distinct
  dashboard.title as "Team Pages",
 dashboard.uid as uid,
   preferences.org_id as org_id,
   concat('https://xxxx/d/',dashboard.uid,'?orgId=',preferences.org_id,'&forcelogin=true') as a
FROM
 Team_Info
 INNER JOIN preferences ON (Team_Info.team_id=preferences.team_id)
 INNER JOIN dashboard ON (preferences.home_dashboard_id=dashboard.id)
  AND (preferences.org_id=dashboard.org_id)
WHERE
  (Team_Info.user_id = '${__user.login}')
  group by Team_Info.user_id,dashboard.title,preferences.org_id

**Reply by ashharrison90** [reply]
mostly complexity. we have many pages - some of these live under static routes so we can switch orgs easily, some of them are dynamic (as with dashboards) where it's not so easy. our priority right now is not regressing previous behaviour whilst we switch to this new navigation. we can (and will) look to make enhancements later.

**Comment by b-neufeld** [comment]
I've got an application that uses grafana-image-renderer to automatically snapshot images of entire dashboards and tweet them out. 

After the 9.5 changes, all my full-dashboard images get an arbitrary number of pixels taller and the images look "off-balance" due to the fat menu on the top of the screen. 

I think the new UI is great for navigability but perhaps there should be a toggle or a switch in the config where the navbar can be collapsed by default for cases like this. In the meantime I will add some application logic to do some cropping. 

**Edit**: actually I think this is a non-issue for me at least. The wasted space has moved from the side of the image to the top. I should have figured out cropping in my application earlier. Carry on!

**Reply by ashharrison90** [reply]
hi @b-neufeld 👋 

i'm not sure how your application works, but you can control visibility of the top navigation bar by using the `kiosk` query param. simply append either `?kiosk=true` or `?kiosk=tv` to the dashboard url visited by the image renderer:

| `kiosk=true` | `kiosk=tv` |
| --- | --- |
| ![image](https://user-images.githubusercontent.com/20999846/234805282-ae934468-4198-4167-8bd2-d17448ad9dd4.png) | ![image](https://user-images.githubusercontent.com/20999846/234805372-c57d3a1e-31bd-47c4-a4b9-5cb1f0bdd0e2.png) |

(please ignore my mess of tabs 😂 )

**Reply by b-neufeld** [reply]
Brilliant! Thanks Ashley. Hadn't been querying the right terms to find this in the docs. Thank you!!

**Comment by the-voidl** [comment]
Is there a way to let the icon navigation on the left side appear again? I don't like the current approach because there's at least one more click required to switch sites. (First pop the menu and then select the desired icon)

**Reply by the-voidl** [reply]
Additionally, it's now 4 clicks to import a dashboard. In the old UI it was 1 mouse hover + 1 click.

I know that I can use Ctrl+K but some times I just have my hand on the mouse and want to do it this way.

**Reply by ashharrison90** [reply]
hi @the-voidl 👋 

no, there's no way to revert to the old navigation. 

rest assured, the left hand menu being clunky is by far the most common piece of feedback we've had. we'll be exploring ways to make this smoother post Grafana 10. 

re: importing a dashboard. we've [added that to the quick add menu](https://github.com/grafana/grafana/pull/65528) in the top right as of v9.5.0: 
![image](https://user-images.githubusercontent.com/20999846/234914868-6a5d8f1b-96ea-4c18-877a-b0cfb53098f1.png)

hope that helps! 🙌

**Reply by the-voidl** [reply]
Got it, but sadly that's still two clicks more than it was before.
Importing dashboards is probably not the most used feature out there but the main pain point for me is that almost every menu entry is one more click far away - while before every important page was reachable through it's corresponding icon on the left had side.

**Reply by amy-super** [reply]
Hi @the-voidl  - I'm interested in your feedback around "extra clicks." Would you be interested in joining me for a 45-minute interview to provide more context? Details are [here](https://www.userinterviews.com/projects/28-0kHr3pw/apply) in this sign-up link. (Edited because I forgot to mention that we'll share a $75 gift card with you as a thanks for your time if you're selected to participate.)

**Comment by meyerder** [comment]
I tend to agree I really would like most of the items on the Right Hand
Side moved to the Left Hand Side.. for those of us with large monitors it
is a PITA and everything "should" be in the same area.. to make it easier
to leverage and convenience aspect

**Comment by KOCIU22** [comment]
Hello Title Name in Panel now is on left side not on middle like before can you add option to chose or back to middle setting by default

![image](https://user-images.githubusercontent.com/78568161/235137294-12bc9bd8-dba4-47ed-a91c-12bfb4710a2a.png)

**Reply by ashharrison90** [reply]
hi @KOCIU22 👋 

that's not related to this piece of work. there is an existing discussion to add that functionality [here](https://github.com/grafana/grafana/discussions/67298), i'm sure @grafana/dashboards-squad are monitoring that for feedback 👍

**Comment by Foorack** [comment]
Hello @ashharrison90 

I have opened a feature request for adding a `feature_toggles` flag to collapse the header by default. We use Grafana for anonymous data visualization for a game, and don't want users to either see the search field nor login button, as they shouldn't do either of those things. Login is restricted to a single admin user, and there is only 1 Dashboard.

![image](https://user-images.githubusercontent.com/5008081/235298678-47e98c84-ee39-4986-8988-a3cc9496f8ee.png)

https://github.com/grafana/grafana/discussions/67573

Do you think this is something which could be easily implemented? It would probably also help those who are not a major fan of the new header UI, by making it less intrusive, as on ultra-wide monitors the new navigation is occupying quite some space.

**Reply by ashharrison90** [reply]
replied in the linked discussion but i'll reply here as well:

we don't plan on adding a config parameter for that behaviour in the immediate term.

i believe we have a mechanism in place currently to accomplish what you want. append `?kiosk=true` or just `?kiosk` to your url. this mode will hide both navigation bars and disable the command palette (press escape to leave this mode) 👍

**Reply by torkelo** [reply]
you can also set kiosk=tv to hide the top search bar

**Comment by jacksongoode** [comment]
Is there a simple way to disable the News panel?

**Reply by ashharrison90** [reply]
@jacksongoode no, there's currently no way to disable the news panel

**Reply by jacksongoode** [reply]
That does seem like something folks should be able to disable with a provisioning flag :)

**Reply by ToMs77130** [reply]
It should be nice indeed, especially for instances used by customers

**Reply by ashharrison90** [reply]
@jacksongoode @ToMs77130 @meyerder we've just merged a PR that exposes a config option to disable the news feed [here](https://github.com/grafana/grafana/pull/69365) which will go live in grafana v10.1.0.

hope that helps! 🙌

**Comment by ohmakai** [comment]
I have a custom app plugin. We used to be able to include the app logo in the left hand nav and include links to relevant dashboards, etc. when hovered. I've updated to [use the PluginPage component](https://grafana.com/docs/grafana/latest/developers/plugins/migration-guide/#switch-to-pluginpage-component), which adapts the included nav items, but they are hidden under the "Apps" menu. I can only see those links if I click through to my custom app page.

The search bar is pretty great, and makes things a bit easier, but I miss the discoverability and ease of use that the old nav provided. Is there any plan to allow custom apps a spot on the top nav?

**Reply by ashharrison90** [reply]
@ohmakai there's no plans currently to let a plugin place itself at the root of the nav tree. part of the problem is **everyone** wants to be at the root of the nav tree, which just isn't feasible with the amount of apps and other functionality grafana now provides 😅 

however, as a server administrator, **you** can configure where a plugin is placed with some server config:
```ini
[navigation.app_sections]
<pluginID> = <sectionID> <sortWeight>
```
so if, for example, you had a plugin with id `myPlugin` and wanted to place it at the root your config would look something like:
```ini
[navigation.app_sections]
myPlugin = root
```

you can find a list of the `sectionID`s [here](https://github.com/grafana/grafana/blob/main/pkg/services/navtree/models.go#L32) and the various `sortWeight`s [here](https://github.com/grafana/grafana/blob/main/pkg/services/navtree/models.go#L14) if you're interested in more granular customisation.

neither of these are particularly well documented at the moment i'm afraid. we haven't really spent much time cleaning up the interface and documenting it yet. it's just been left there as an escape hatch for people who really want to move plugins around in the nav tree.

we've also discussed in the past letting users customise their navigation menu through preferences or some other option, so we may explore that in the future as well 👍

**Comment by pgloader** [comment]
Is there a way to restore the sidebar of 9.1. 6 ?

**Reply by ashharrison90** [reply]
@pgloader no, there's no way to restore the old sidebar. we'll be continuing to iterate on the new navigation moving forward.

**Comment by dbeast-co** [comment]
We use Grafana with a community and custom application plugins, and in the new version, the additional clicks are really not user-friendly. 
I saw the answer about adding the app to the root menu, but this is still two clicks and is hidden from the users at the start.

I think, that it could be helpful to add the configuration variable in the plugin.json, which provides the option to add the application plugin in the top menu, or somewhere else, with one-click access to the application.

**Comment by ajacques** [comment]
One thing I miss from the 9.4.x UI is the left nav panel is always present. I don't want it to be hidden behind a hamburger menu. My screen is wide and I have plenty of space. Keep the icons fixed to the left like it used to please.

**Reply by amy-super** [reply]
Hey @ajacques - we're evaluating a further evolution of the design that lets users pin the menu open. Please stay tuned!

**Comment by meyerder** [comment]
It appears that there is a considerable number of individuals who may be hesitant to transition to the new style. For those of us who have numerous users, this represents a significant change that will require substantial effort and challenges to modify user behavior. The approach being taken seems quite strict, which can be disheartening for many of us.

While this discussion is a positive indication, it is evident that the decision has been made and will be implemented. Personally, I am curious about the initial requests that led to this decision. Were they technical in nature or were they made by a larger group of people?

For instance, one seemingly straightforward request is the ability to disable news. It seems somewhat questionable to continuously display it to paying customers. Perhaps it would be more understandable if it was limited to the free version, as a means to generate more web page visits and potentially increase revenue.

From my perspective, it is evident that Grafana is focused on the future and aims to position the product for quick adaptability. However, taking such a rigid stance and neglecting to address existing issues in prior versions, alongside the current changes, does not align with what I consider to be a path for successful company growth.

**Reply by amy-super** [reply]
Hey @meyerder ,
Thanks for the feedback! We are continually looking for ways to make Grafana's new navigation better, so it's extra helpful for us to see specific requests like hiding the news panel (which we're discussing internally and will reply about separately). 

Regarding the reason for the shift as a whole, we received feedback across our user base that the left-side icons were difficult to learn and use, running off of the page (especially for people using Grafana Cloud), and that generally too many things were being presented on the "front page" of Grafana. @yaelleC  is going to be presenting some more details, including how we are measuring the redesign's impact and usability, at [GrafanaCON](https://grafana.com/about/events/grafanacon/2023/session/new-grafana-10-product-navigation/). Highly recommend checking out her talk or the recording afterwards. 

In addition, I'm currently running a research/interview study specifically about the feedback we've heard the most - that people feel like we introduced too many clicks into their workflow. We don't in any way feel like this work is "done" but will continue to be iterated upon. So please keep the specific feedback coming!

**Reply by meyerder** [reply]
I apologize, but I find it concerning that the redesign was launched as a stable release before gathering feedback. It seems counterintuitive to seek feedback after the fact. Additionally, it's frustrating that bug fixes for prior versions are only being addressed in version 9.5+, which means we are forced to accept the new changes, including features like the news panel and new pages, without the ability to opt out. It would have been more ideal to introduce such changes as part of a feature enablement aspect that allows users to enable/disable them until after the research/interview study that you are doing and allows for capture of simple items like one of my prior items (not being able to change Orgs) that would have been caught before moving it into a GA release verse us capturing new bugs and then it being decided on if it really is a bug or not.  Or even if the feedback was for people running Grafana Cloud to implement it only for Grafana Cloud vs on Prem instances

**Reply by ashharrison90** [reply]
hi again @meyerder 👋 

our intention isn't to be strict at all. it's just not feasible for us to support 2 completely different navigation patterns and continue to iterate in a positive direction. in that sense we need to flip the switch at some point so we can focus our efforts into making the new navigation experience the best possible.

you can disable the new navigation in 9.5.x by setting the `topnav` feature toggle to false. from v10 that will no longer be an option.

> I apologize, but I find it concerning that the redesign was launched as a stable release before gathering feedback. It seems counterintuitive to seek feedback after the fact.

think this is worth addressing directly: i don't believe that's correct. at all stages of the process we've been asking for and evaluating feedback:
  - we ran external user tests during development
    - these showed positive results compared to the existing navigation
  - we released this [behind a feature flag as early as grafana v9.3](https://grafana.com/blog/2022/11/29/grafana-9.3-release/) asking for feedback
    - that's what this discussion was created for in the first place
  - we've been taking internal feedback from the hundreds of grafanistas that use grafana daily
  
@amy-super is currently running some **additional** user tests to evaluate some prototypes we're hoping will alleviate the most common pain point we've heard around the hamburger menu leading to additional clicks. but these are not the first round of user tests/feedback we've done. hope that clears up any confusion 😅 

we are super grateful for everyone that has provided feedback so far (yourself included!) 🙌 as amy said, we'll continue to iterate and improve the experience going forward. 👍

**Reply by meyerder** [reply]
@ashharrison90 Can you please confirm that in 9.5x it can be disabled?   On Apr 15 (Shortly after 9.5 came out) I asked if it could be disabled (in this thread) and was replied "this will be the default navigation going forward so there will be no way to disable "  This was when 9.5 First came out.  I was under the impression maybe incorrectly that as of 9.5x there was no option but to have this..   If this is the case much of what I stated initially I take back and I will re-evaluate the 9.5x branch..

**Reply by ashharrison90** [reply]
@meyerder yes it can be disabled in 9.5.x, but from v10 it will be the only option. apologies for the confusion.

**Comment by gbeyer3** [comment]
This concerns navigation within dashboards.  This request was first made years ago, has been concurred with by many users, but has not been implemented.  Variable selectors need to remain visible and usable at screen top when scrolling to lower panels.   In dashboards which are longer than one screen height, variable selectors scroll off the page, when scrolling down to lower panels.   Workflow is extremely frustrating:  Scroll down to view lower panels, scroll BACK top the top to select a different variable, then scroll back DOWN again to see the change in the panel resulting from the variable change.  Miss-selected variable, or need to double-check what was selected, or need to tweak/filter what I am seeing via the variables - results in repeating the same scroll up/scroll down, scroll up/scroll down process over and over.    WHEN will this be implemented?
